src/Main.tsx

```diff
@@ -63,12 +64,21 @@ import arrowLeftImg from "../images/imgarrowlefticon.svg";
 import { KOCommentEnd, KOCommentIfStart } from "./koComment";
 import { useConfig } from "./hooks/useConfig";
 import { useKnockoutExplorer } from "./hooks/useKnockoutExplorer";
+import { NotificationConsoleComponent } from "./Explorer/Menus/NotificationConsole/NotificationConsoleComponent";
 
 initializeIcons();
 
 const App: React.FunctionComponent = () => {
+  const [isNotificationConsoleExpanded, setIsNotificationConsoleExpanded] = useState(false);
+  const [notificationConsoleData, setNotificationConsoleData] = useState(undefined);
+  const [inProgressMessageIdToBeDeleted, setInProgressMessageIdToBeDeleted] = useState("");
```

Comment:
If I'm correct, `App` wants to delete an item in the `NotificationConsole` array, so it queues this, for the notification console to execute the delete and reset this to `undefined` when done?
I kind of like this pattern because it keeps things encapsulated, but the React purists will probably say that this is a functional programming anti-pattern and that if we want to delete an item, we should be able to do so synchronously by calling a function or directly altering the array. Which probably means bubbling up the `NotificationConsole` `consoleData` array into `App`. We are already exposing the id's in `App`, so we're not completely hiding `ConsoleData`. @southpolesteve What are your thoughts on this?

Reply:
I'll look into refactoring this later. I've added a TODO in the comments.